gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Policy.java

```diff
@@ -0,0 +1,389 @@
+      return Objects.hash(memberType, emailOrDomain);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof Member && Objects.equals(this.memberType, ((Member) obj).memberType)
+          && Objects.equals(this.emailOrDomain, ((Member) obj).emailOrDomain);
+    }
+  }
+
+  /**
+   * Represents an IAM policy binding
+   */
+  public static class Binding implements Serializable {
+
+    private static final long serialVersionUID = -8493421092718338925L;
+    private final RoleType role;
+    private final List<Member> members;
+
+    public static class Builder {
+      private RoleType role;
+      private List<Member> members;
+
+      Builder() {
+        members = new ArrayList<Member>();
+      }
+
+      public Builder role(RoleType role) {
+        this.role = role;
+        return this;
+      }
+
+      public Builder members(List<Member> members) {
+        this.members = checkNotNull(members);
+        return this;
+      }
+
+      public Builder clearMembers() {
+        this.members = new ArrayList<>();
+        return this;
+      }
+
+      public Builder addMember(Member member) {
+        this.members.add(member);
+        return this;
+      }
+
+      public Builder removeMember(Member member) {
+        this.members.remove(member);
+        return this;
+      }
+
+      public Binding build() {
+        return new Binding(role, members);
+      }
+    }
+
+    private Binding(RoleType role, List<Member> members) {
+      this.role = role;
+      ImmutableList.Builder<Member> membersListBuilder = new ImmutableList.Builder<>();
+      for (Member member : members) {
+        membersListBuilder.add(member);
+      }
+      this.members = membersListBuilder.build();
+    }
+
+    public static Binding binding(RoleType role, List<Member> members) {
+      return new Binding(role, members);
+    }
+
+    public RoleType role() {
+      return role;
+    }
+
+    public List<Member> members() {
+      return members;
+    }
+
+    public static Builder builder() {
+      return new Builder();
+    }
+
+    public Builder toBuilder() {
+      List<Member> mutableMembers = new ArrayList<>();
+      for (Member member : members) {
+        mutableMembers.add(member);
+      }
+      return new Builder().role(role).members(mutableMembers);
+    }
+
+    com.google.api.services.cloudresourcemanager.model.Binding toPb() {
+      com.google.api.services.cloudresourcemanager.model.Binding bindingPb =
+          new com.google.api.services.cloudresourcemanager.model.Binding();
+      if (role != null) {
+        bindingPb.setRole("roles/" + role.toString().toLowerCase());
+      }
+      List<String> membersPb = new ArrayList<>(members.size());
+      for (Member member : members) {
+        if (member.emailOrDomain() != null) {
+          membersPb.add(member.type().prefix() + member.emailOrDomain());
+        } else {
+          membersPb.add(member.type().prefix());
+        }
+      }
```

Comment:
Wouldn't it be better to put this logic in a `toPb` method inside `Member`? Something like:
```java
public String toPb() {
  return emailOrDomain != null ? memberType.prefix() + emailOrDomain : memberType.prefix();
}
```